File: python_practice/1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#
from typing import List

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # use dict as our value container
        num_dic = dict()
        for index, num in enumerate(nums):
            # see if in hash keys
            if target - num in num_dic:
                return [index, num_dic[target - num]]
            num_dic[num] = index
        return []
                
        
        
# @lc code=end

File: cpp/1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */
#include <unordered_map>
#include <vector>
using namespace std;
// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> mappings;
        int y;
        for (int i = 0; i < nums.size(); i++)
        {
            y = target - nums[i];
            if (mappings.find(y) != mappings.end())
            {
                return {i, mappings[y]};
            }
            else
            {
                mappings[nums[i]] = i;
            }
        }
        return {}; // Return empty vector if no solution found
    }
};
// @lc code=end

